src/main/java/io/github/oliviercailloux/collaborative_exams/controller/NewSameAbility.java

```diff
@@ -46,13 +41,42 @@ public String newSameAbility(MultivaluedMap<String, String> form) throws Excepti
 			throw new Exception("You indicated the same id Question for both of the questions.");
 
 		}
-
-		SameAbility s = new SameAbility(questionService.findQuestion(idQuestion1),
-				questionService.findQuestion(idQuestion2), personService.findPerson(idAuthor));
+		
+		/*
+		 * On positionne d'abord la question avec l'id le plus petit afin d'optimiser la recherche
+		 * 
+		 */
+		
+		SameAbility s;
+		
+		if(idQuestion1 < idQuestion2)
+		{
+			s = new SameAbility(questionService.findQuestion(idQuestion1),
+			questionService.findQuestion(idQuestion2), personService.findPerson(idAuthor));
+		}
+		else
+		{
+			s = new SameAbility(questionService.findQuestion(idQuestion1),
+			questionService.findQuestion(idQuestion2), personService.findPerson(idAuthor));
+		}
```

Comment:
- Formater le code pour suivre les conventions habituelles.
- Éviter cette répétition inélégante. Affecter d’abord q1 et q2 dans le bon ordre, puis invoquer le même code une seule fois.